SyntheticsRunTestsTask/task.ts

```diff
@@ -1,47 +1,49 @@
 import * as path from 'path'
 
 import * as task from 'azure-pipelines-task-lib/task'
-import * as util from 'azure-pipelines-tasks-packaging-common/util'
 
-import {synthetics} from '@datadog/datadog-ci'
-
-import {renderResults} from './process-results'
-import {reportCiError} from './report-ci-error'
 import {getReporter, resolveConfig} from './resolve-config'
+import {synthetics} from '@datadog/datadog-ci'
 
 async function run(): Promise<void> {
   task.setResourcePath(path.join(__dirname, 'task.json'))
   synthetics.utils.setCiTriggerApp('azure_devops_task')
 
   const reporter = getReporter()
   const config = await resolveConfig(reporter)
+  const startTime = Date.now()
+
+  let results: synthetics.Result[]
+  let summary: synthetics.Summary
 
   try {
-    const startTime = Date.now()
-    const {results, summary} = await synthetics.executeTests(reporter, config)
-    const resultSummary = renderResults({config, reporter, results, startTime, summary})
-    if (
-      resultSummary.criticalErrors > 0 ||
-      resultSummary.failed > 0 ||
-      resultSummary.timedOut > 0 ||
-      resultSummary.testsNotFound.size > 0
-    ) {
-      task.setResult(task.TaskResult.Failed, `Datadog Synthetics tests failed : ${printSummary(resultSummary)}`)
-    } else {
-      task.setResult(task.TaskResult.Succeeded, `Datadog Synthetics tests succeeded : ${printSummary(resultSummary)}`)
-    }
+    ;({results, summary} = await synthetics.executeTests(reporter, config))
   } catch (error) {
-    if (error instanceof synthetics.CiError) {
-      reportCiError(error, reporter)
-    } else {
-      util.logError(`Internal error: ${String(error)}`)
+    synthetics.utils.reportExitLogs(reporter, config, {error})
+
+    const exitReason = synthetics.utils.getExitReason(config, {error})
+    if (exitReason !== 'passed') {
+      task.setResult(task.TaskResult.Failed, 'Running Datadog Synthetics tests failed.')
     }
-    task.setResult(task.TaskResult.Failed, 'Running Datadog Synthetics tests failed.')
+    return
   }
-}
 
-export const printSummary = (summary: synthetics.Summary): string =>
-  `criticalErrors: ${summary.criticalErrors}, passed: ${summary.passed}, failedNonBlocking: ${summary.failedNonBlocking}, failed: ${summary.failed}, skipped: ${summary.skipped}, notFound: ${summary.testsNotFound.size}, timedOut: ${summary.timedOut}`
+  const orgSettings = await synthetics.utils.getOrgSettings(reporter, config)
+
+  synthetics.utils.renderResults({
+    config,
+    orgSettings,
+    reporter,
+    results,
+    startTime,
+    summary,
+  })
+
+  const exitReason = synthetics.utils.getExitReason(config, {results})
+  if (exitReason !== 'passed') {
+    task.setResult(task.TaskResult.Failed, 'Datadog Synthetics tests failed.')
```

Comment:
It looks like the `setResult` used to have the `printSummary(resultSummary)` in the message - can we keep that functionality in some way?

Reply:
For example, the github action has different logic for the summary:

https://github.com/DataDog/synthetics-ci-github-action/blob/ea9a637d4be4fc9d31212639baf5f86ebcf84618/src/main.ts#L37-L44

Reply:
Updated `printSummary` to align with the github action.